src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -12,19 +22,185 @@ namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+            {
+                return solution;
+            }
+
+            foreach (var referenceLocationGroup in references.Single().Locations.GroupBy(r => r.Document))
+            {
+                // Get document in current solution
+                var document = solution.GetDocument(referenceLocationGroup.Key.Id);
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                // Compute replacements
+                var editor = new SyntaxEditor(root, solution.Workspace);
+                foreach (var referenceLocation in referenceLocationGroup)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var referenceNode = root.FindNode(referenceLocation.Location.SourceSpan, getInnermostNodeForTie: true);
+                    if (referenceNode != null)
+                    {
+                        var operation = GetOperationWalkingUpParentChain(referenceNode, semanticModel);
+                        SyntaxNode nodeToReplaceOpt = null;
+                        switch (operation)
+                        {
+                            case IMemberReferenceOperation memberReference:
+                                if (IsReplacableOperation(memberReference.Instance))
+                                {
+                                    nodeToReplaceOpt = GetSyntaxNodeToReplace(memberReference);
+                                }
+
+                                break;
 
+                            case IInvocationOperation invocation:
+                                if (IsReplacableOperation(invocation.Instance))
+                                {
+                                    nodeToReplaceOpt = GetExpressionOfInvocation(invocation.Syntax);
+                                }
+
+                                break;
+                        }
+
+                        if (nodeToReplaceOpt != null)
+                        {
+                            // Fetch the symbol for the node to replace - note that this might be
+                            // different from the original symbol due to generic type arguments.
+                            var symbolInfo = semanticModel.GetSymbolInfo(nodeToReplaceOpt, cancellationToken);
+                            if (symbolInfo.Symbol == null)
+                            {
+                                continue;
+                            }
+
+                            SyntaxNode memberName;
+                            var typeArgumentsOpt = GetTypeArguments(referenceNode);
+                            memberName = typeArgumentsOpt != null ?
+                                editor.Generator.GenericName(symbolInfo.Symbol.Name, typeArgumentsOpt) :
+                                editor.Generator.IdentifierName(symbolInfo.Symbol.Name);
```

Comment:
should this be the fully qualified name in case that the type is not imported?

Reply:
Sorry, left this at wrong location. I was talking about the `expression` generated below. Also it seems SyntaxGenerator has some logic about this, I'm not sure how it works though.

---
In reply to: [246142293](https://github.com/dotnet/roslyn-analyzers/pull/1982#discussion_r246142293) [](ancestors = 246142293)

Reply:
`editor.Generator.TypeExpression(symbolInfo.Symbol.ContainingType)` invocation below generates a fully qualified type name, with simplifier annotation.

Reply:
> Sorry, left this at wrong location. I was talking about the expression generated below. Also it seems SyntaxGenerator has some logic about this, I'm not sure how it works though.

SyntaxGenerator.TypeExpression walks up the containing type/namespace chain to always generate a fully qualified name with simplifier annotations: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs,185